myokit/formats/diffsl/_ewriter.py

```diff
@@ -0,0 +1,205 @@
+
+    # def _ex_log(self, e):
+
+    def _ex_log10(self, e):
+        # Log10(a) = Log(a, 10.0) -> '(log(a) / log(10.0))'
+        return super()._ex_log(Log(e[0], Number(10)))
+
+    # def _ex_minus(self, e):
+    # def _ex_multiply(self, e):
+    # def _ex_plus(self, e):
+    # def _ex_power(self, e):
+    # def _ex_prefix_minus(self, e):
+    # def _ex_prefix_plus(self, e):
+    # def _ex_quotient(self, e):
+    # def _ex_remainder(self, e):
+    # def _ex_sin(self, e):
+    # def _ex_sqrt(self, e):
+    # def _ex_tan(self, e):
+
+    # -- Conditional operators
+
+    def _ex_and(self, e):
+        # (a and b) == a * b, where a, b are in {0, 1}
+        return f'{self.ex(e[0])} * {self.ex(e[1])}'
+
+    def _ex_equal(self, e):
+        # (a == b) == heaviside(a - b) * heaviside(b - a)
+        return self.ex(And(MoreEqual(e[0], e[1]), LessEqual(e[0], e[1])))
+
+    def _ex_less(self, e):
+        # (a < b) == 1 - heaviside(a - b)
+        return self.ex(Not(MoreEqual(e[0], e[1])))
+
+    def _ex_less_equal(self, e):
+        # (a <= b) == heaviside(b - a)
+        return f'heaviside({self.ex(e[1])} - {self.ex(e[0])})'
+
+    def _ex_more(self, e):
+        # (a > b) == 1 - heaviside(b - a)
+        return self.ex(Not(LessEqual(e[0], e[1])))
+
+    def _ex_more_equal(self, e):
+        # (a >= b) == heaviside(a - b)
+        return f'heaviside({self.ex(e[0])} - {self.ex(e[1])})'
+
+    def _ex_not(self, e):
+        # Simplify expressions with double not's.
+        # Examples:
+        # not(not(a)) -> (1 - (1 - a)) == a
+        # not(a or b) -> (1 - (1 - (1 - a) * (1 - b))) == (1 - a) * (1 - b)
+        if isinstance(e[0], Not):
+            # not(not(a)) == a
+            return self.ex(e[0][0])
+
+        if isinstance(e[0], NotEqual):
+            # not(not(a == b)) == (a == b)
+            return self.ex(Equal(e[0][0], e[0][1]))
+
+        if isinstance(e[0], Less):
+            # not(a < b) == (a >= b)
+            return self.ex(MoreEqual(e[0][0], e[0][1]))
+
+        if isinstance(e[0], More):
+            # not(a > b) == (a <= b)
+            return self.ex(LessEqual(e[0][0], e[0][1]))
+
+        if isinstance(e[0], Or):
+            # not(a or b) == not(a) and not(b)
+            return self.ex(And(Not(e[0][0]), Not(e[0][1])))
+
+        # not(a) == (1 - a), where a is in {0, 1}
+        return f'(1 - {self.ex(e[0])})'
+
+    def _ex_not_equal(self, e):
+        # (a != b) == 1 - heaviside(a - b) * heaviside(b - a)
+        return self.ex(Not(Equal(e[0], e[1])))
+
+    def _ex_or(self, e):
+        # a or b == not(not(a) and not(b)), where a, b are in {0, 1}
+        return self.ex(Not(And(Not(e[0]), Not(e[1]))))
+
+    # -- Conditional expressions
+
+    def _ex_if(self, e):
+        # General expression:
+        # if(_i, _t, _e) -> _t * _i + _e * not(_i), where _i is in {0,1}
+
+        # Shorten expression if _t == _e
+        # if(_i, _t, _t) -> _t
+        if e._t == e._e:
+            return self.ex(e._t)
```

Comment:
This should be removed. Two reasons:

1. "if()" as a function has higher operator precedence than e.g. "1 + 1" so this could turn `5 * if(...)` into `5 * 1 + 1`
2. This is not the right place for these optimisations. If you really need these, implement them in the symbolic stage before exprt